worktree_status.go

```diff
@@ -409,6 +585,43 @@ func (w *Worktree) deleteFromFilesystem(path string) error {
 	return err
 }
 
+// RemoveGlob given a glob pattern removes all the matching files content and
+// all his sub-directories recursively in the worktree to the index.
+func (w *Worktree) RemoveGlob(pattern string) error {
+	files, err := util.Glob(w.Filesystem, pattern)
+	if err != nil {
+		return err
+	}
+
+	if len(files) == 0 {
```

Comment:
I'm curious here how we might use this method for a paths that might still exist in the repository or index, but might have already been removed from the working directory. Will this check cause this method to return an error without attempting to remove them from the index?

Example:
```
$ git init
$ touch a; git add a; git commit -m first
[master (root-commit) 5d36496] first
 1 file changed, 0 insertions(+), 0 deletions(-)
 create mode 100644 a
$ rm a; git status
On branch master
Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git checkout -- <file>..." to discard changes in working directory)

        deleted:    a

no changes added to commit (use "git add" and/or "git commit -a")
$ git rm -- a*; git status
rm 'a'
On branch master
Changes to be committed:
  (use "git reset HEAD <file>..." to unstage)

        deleted:    a

```

Reply:
@mcuadros this seems like a bug to me. Can you confirm?

In `git add`, you can modify entries in the index even if the glob matches no paths in the current working directory. But this check looks like it would return an error before trying to update the index if the glob has no matches in the working directory.
```
$ git init
$ touch a; git add a
$ rm a; ls a*
$ git add -- a*; echo $?
0
```

Reply:
I think this is a very complex behavior, the goal of this PR was speed up multiple `Add` calls.